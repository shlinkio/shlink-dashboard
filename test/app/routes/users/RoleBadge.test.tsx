import { render } from '@testing-library/react';
import type { Role } from '../../../../app/entities/User';
import { RoleBadge } from '../../../../app/routes/users/RoleBadge';
import { checkAccessibility } from '../../../__helpers__/accessibility';

describe('<RoleBadge />', () => {
  const setUp = (role: Role) => render(<RoleBadge role={role} />);
  const testCases = [
    { role: 'admin' as const },
    { role: 'advanced-user' as const },
    { role: 'managed-user' as const },
  ];

  it.each(testCases)('passes a11y checks', ({ role }) => checkAccessibility(setUp(role)));

  it.each(testCases)('renders expected markup', ({ role }) => {
    const { container } = setUp(role);
    expect(container).toMatchSnapshot();
  });
});
