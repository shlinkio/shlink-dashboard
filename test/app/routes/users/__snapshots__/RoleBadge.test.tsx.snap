// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<RoleBadge /> > renders expected markup 1`] = `
<div>
  <div
    class="rounded-sm px-1 inline-block font-bold whitespace-nowrap bg-green-600 text-white"
  >
    admin
  </div>
</div>
`;

exports[`<RoleBadge /> > renders expected markup 2`] = `
<div>
  <div
    class="rounded-sm px-1 inline-block font-bold whitespace-nowrap bg-gray-500 text-white"
  >
    advanced user
  </div>
</div>
`;

exports[`<RoleBadge /> > renders expected markup 3`] = `
<div>
  <div
    class="rounded-sm px-1 inline-block font-bold whitespace-nowrap bg-gray-500 text-white"
  >
    managed user
  </div>
</div>
`;
